vtex-sites/storecomponents.store
# feat: fix product click event

## What's the purpose of this pull request?
This PR fixes the product click event by updating the dependencies and passing on the position to the updated useLink hook.

## How it works? 
The position received is now sent to the useLink hook. The hook also grabs information from the SearchContext and sets the pageType and term automatically (but they can be overridden).

## How to test it?
Run this on your console to see the events being sent:
```js
window.addEventListener("message", function (event) {
  if (event.data?.data?.type?.startsWith("vtex:")) {
    console.log(event.data.data);
  }
});
```

Now click on products (the easiest way is to click the add to cart button). Check if the product position is right, if the `fullText` page sends the appropriate pageType and term, and if category/brand pages send the pageType as `nonFullText` and `null` term. On other pages such as the home, the pageType should be `other`.
